### PR TITLE
Laboratory work #8, Koykova Aleksandra - 20FPL2

### PR DESCRIPTION
**Pull request by KoykovaAleksandra**

### CODE REVIEW

File: lab_8_llm/main.py [hunk 1]
```diff
@@ -34,6 +37,7 @@ class RawDataImporter(AbstractRawDataImporter):
     """
     A class that imports the HuggingFace dataset.
     """
+    _raw_data: DataFrame
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: Are you sure this announcement is necessary?

File: lab_8_llm/main.py [hunk 2]
```diff
@@ -43,6 +47,15 @@ def obtain(self) -> None:
         Raises:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
+        self._raw_data = load_dataset(self._hf_name,
+                                      split='train').to_pandas()
+
+    @property
+    def get_raw_data(self) -> DataFrame:
```
Comment (SalnikovIgor) [inline lab_8_llm/main.py]: And this?

File: lab_7_llm/main.py [hunk 3]
```diff
@@ -2,28 +2,24 @@
 Neural machine translation module.
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Checkout the files from 7 lab to the master

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+datasets==2.16.1
```
Comment (SalnikovIgor) [inline requirements.txt:2]: Add `--extra-index-url https://download.pytorch.org/whl/cpu`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 4 above)
Comment (KoykovaAleksandra) [reply requirements.txt:2]: Done

File: lab_8_llm/main.py (hunk 1 above)
Comment (KoykovaAleksandra) [reply lab_8_llm/main.py]: If I remove this announcement, then I get an error in mypy checks:
Снимок экрана 2024-03-08 231054

File: lab_8_llm/main.py (hunk 2 above)
Comment (KoykovaAleksandra) [reply lab_8_llm/main.py]: The same situation(

File: lab_8_llm/main.py (hunk 1 above)
Comment (SalnikovIgor) [reply lab_8_llm/main.py]: Need to add a check: `_raw_data` containt `None`? If yes, then rise error

File: lab_8_llm/main.py (hunk 2 above)
Comment (SalnikovIgor) [reply lab_8_llm/main.py]: Comment above

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix comments

File: lab_8_llm/main.py (hunk 1 above)
Comment (KoykovaAleksandra) [reply lab_8_llm/main.py]: Done!)

File: lab_8_llm/main.py (hunk 2 above)
Comment (KoykovaAleksandra) [reply lab_8_llm/main.py]: Done!)

File: lab_7_llm/main.py (hunk 3 above)
Comment (KoykovaAleksandra) [reply lab_7_llm/main.py]: Done!)

File: lab_7_llm/main.py [hunk 5]
```diff
@@ -192,4 +192,4 @@ def run(self) -> dict | None:
 
         Returns:
             dict | None: A dictionary containing information about the calculated metric
-        """
+        """
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Again lab 7 😑

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/main.py (hunk 5 above)
Comment (KoykovaAleksandra) [reply lab_7_llm/main.py]: Finally, done!))

**Pull request closed without merging**